### ISSUE TITLE
Clearly dominant color is not extracted

### ISSUE THREAD

**Issue by crishoj** [ksubileau/color-thief-php#36]
Consider the following image:

![margrethe-hjort-hay-2018-margrethes-perler-indbundet-bog](https://user-images.githubusercontent.com/20393/42969109-5287e05a-8bcf-11e8-9e62-130ab338d788.jpg)

A visual inspection leads me to believe that the dominant color is yellow:

<img width="243" alt="image" src="https://user-images.githubusercontent.com/20393/42969147-7465ac02-8bcf-11e8-8663-a1eec11402a4.png">

However, this color is not among the dominant colors extracted by ColorThief:

<img width="229" alt="image" src="https://user-images.githubusercontent.com/20393/42969173-8dc211b8-8bcf-11e8-9b58-fac25b16406e.png">

At first I thought the bright yellow was misidentified as white-ish and thus discarded. However, in a [test case](https://github.com/crishoj/color-thief-php/blob/531881538934e5de21973fa520e54078af2ed742/tests/ColorThiefTest.php#L142), `isNonWhite` returns true for `(253, 230, 44)`...

**Comment by crishoj** [comment]
The problem seems to lie in color extraction:

```
There were 2 failures:

1) ColorThief\Image\Adapter\Test\GmagickImageAdapterTest::testGetPixelColorFromCmykJpg
Failed asserting that two objects are equal.
--- Expected
+++ Actual
@@ @@
 stdClass Object (
-    'red' => 255
-    'green' => 229
-    'blue' => 44
+    'red' => 0
+    'green' => 26
+    'blue' => 211
     'alpha' => 0
 )

/Users/christian/Code/color-thief-php/tests/Image/Adapter/BaseImageAdapterTest.php:186

2) ColorThief\Image\Adapter\Test\ImagickImageAdapterTest::testGetPixelColorFromCmykJpg
Failed asserting that two objects are equal.
--- Expected
+++ Actual
@@ @@
 stdClass Object (
-    'red' => 255
-    'green' => 229
-    'blue' => 44
-    'alpha' => 0
+    'red' => 0.0
+    'green' => 26.0
+    'blue' => 211.0
+    'alpha' => 0.0
 )

/Users/christian/Code/color-thief-php/tests/Image/Adapter/BaseImageAdapterTest.php:186
```

**Comment by ksubileau** [comment]
Hi,
Indeed, [the isNonWhite function](https://github.com/ksubileau/color-thief-php/blob/b726455d58570e4e0571d108e7c5f2dbb76250c2/lib/ColorThief/ColorThief.php#L237) simply return true when all channel values are greater than a constant threshold. Nothing very magical, so it's not surprising that this yellow color is not considered as white-ish.

Looking at you last comment, this seems to be a problem/bug with the underlying library (Imagick/Gmagick), or a misuse of it, as the [getPixelColor function](https://github.com/crishoj/color-thief-php/blob/b726455d58570e4e0571d108e7c5f2dbb76250c2/lib/ColorThief/Image/Adapter/ImagickImageAdapter.php#L80) is simply a wrapper around it.

Could you try to extract the pixel color outside of ColorThief, using only Imagick functions, and see if you get the same wrong pixel values ? The code should be something like this (pseudo-code) : 
```php
$resource = new Imagick("/path/to/image.jpg");
$pixel = $resource->getImagePixelColor(20, 20);
$colorArray = $pixel->getColor(true);
var_dump($colorArray);
```

**Comment by crishoj** [comment]
> Could you try to extract the pixel color outside of ColorThief, using only Imagick functions, and see if you get the same wrong pixel values ?

```
php > $resource = new Imagick("tests/images/bookcover.jpg");
php > $pixel = $resource->getImagePixelColor(20, 20);
php > $colorArray = $pixel->getColor(true);
php > var_dump($colorArray);
php shell code:1:
array(4) {
  'r' =>
  double(0)
  'g' =>
  double(0.10196078431373)
  'b' =>
  double(0.82745098039216)
  'a' =>
  double(1)
}
php > var_dump([$colorArray['r'] * 255, $colorArray['g'] * 255, $colorArray['b'] * 255]);
php shell code:1:
array(3) {
  [0] =>
  double(0)
  [1] =>
  double(26)
  [2] =>
  double(211)
}
```

**Comment by crishoj** [comment]
[Transforming the colorspace to SGRB](https://github.com/crishoj/color-thief-php/commit/f9f215c95b5740c3b9a26bacfffa00c31a9a38f1) fixes the problem with Imagick, but seems to break other test cases with Gmagick.

**Comment by ksubileau** [comment]
Hmm ok, so we need to investigate more to find the right way to get pixel color of CMYK images with Imagick/Gmagick

**Comment by crishoj** [comment]
The trouble with Gmagick was caused by using SRGB rather than RGB. Tests now pass. I'll send a pull request.

**Comment by ksubileau** [comment]
Closed by PR #37
